### PR TITLE
V2: use seed and pytorch_seed appropriately

### PR DESCRIPTION
**Pull request by kevingreenman**
## Description
Fixes the usage of the `seed` and `pytorch-seed` CLI arguments, the latter of which was not used at all.

I've renamed `seed` to `data-seed` to better reflect its purpose. This argument was already used for data splitting, but I've now added it to the train dataloader as well. `shuffle` will still be `True` by default for the train dataloader (and `False` for the val and test dataloaders), but users who run the same training job multiple times without specifying the seed will now get reproducible results. This is how we did it in v1.

## Questions
* Should `seed` be `None` by default in `MolGraphDataLoader`? That's what we have now. If we're going to have the seed for the train dataloader be 0 by default when using the CLI (matching v1), should we also update this default to 0?

### CODE REVIEW

**Review by KnathanM — COMMENTED** [review]
To your question about the seed in the `MolGraphDataLoader`, I think we should set the default to be 0 instead of None (and remove the `seed is not None`). This means using this dataloader will never use the pytorch randomness generator, which allows us to separate data splitting randomness from model initialization randomness (if that is what we want). I discussed this more in #661.

File: chemprop/cli/train.py [hunk 1]
```diff
@@ -450,23 +458,22 @@ def add_train_args(parser: ArgumentParser) -> ArgumentParser:
...
-    )
-
-    parser.add_argument(  # TODO: do we need this?
+    parser.add_argument(
         "--pytorch-seed",
         type=int,
         default=0,
         help="Seed for PyTorch randomness (e.g., random initial weights).",
     )
+    split_args.add_argument(
```
[the remark this answers is not in the document]
Comment (kevingreenman) [reply chemprop/cli/train.py]: good catch, done

**Review by KnathanM — APPROVED** [review]
